velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -214,13 +251,18 @@ SubstraitVeloxPlanConverter::processSortField(
 core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
     const ::substrait::FilterRel& filterRel) {
   auto childNode = convertSingleInput<::substrait::FilterRel>(filterRel);
-  const auto& inputType = childNode->outputType();
-  const auto& sExpr = filterRel.condition();
 
-  return std::make_shared<core::FilterNode>(
+  auto noEmitNode = std::make_shared<core::FilterNode>(
       nextPlanNodeId(),
-      exprConverter_->toVeloxExpr(sExpr, inputType),
+      exprConverter_->toVeloxExpr(
+          filterRel.condition(), childNode->outputType()),
       childNode);
+
+  if (filterRel.has_common()) {
```

Comment:
This if block repeats multiple time. Consider folding it into processEmit?

```
return addProjectionIfNeeded(filterRel.common(), std::move(filterNode));
```

Reply:
May I ask what should be the proposed change inside the method, given that `has_common()` checks whether there is a `common` or not.

Reply:
I'm thinking that this code can be simplified to:

```
return processEmit(filterRel, std::move(filterNode));
```

and processEmit would be modified to

```
core::PlanNodePtr processEmit(
      const ::substrait::Rel& rel,
      const core::PlanNodePtr& noEmitNode) {
    if (!rel.has_common()) {
       return noEmitNode;
    }
    
    ...
 )
```

Reply:
@mbasmanova AFAIK `substrait::Rel` doesn't have a common, but `ReadRel` or `ProjectRel` do have a `common`. 
And they are not extended from `Rel`

Example: 
https://github.com/substrait-io/substrait/blob/main/proto/substrait/algebra.proto#L148-L153
https://github.com/substrait-io/substrait/blob/main/proto/substrait/algebra.proto#L354-L373

Reply:
I see. Thank you for clarifying.

Reply:
Of course, I took the template route earlier because of this reason. And I understand it is not a good fit for performance and readability.